### DISCUSSION TITLE
Json Uploader / Splitter

### DISCUSSION THREAD

**Discussion by MehrCurry** [FlowiseAI/Flowise#1086] [Q&A]
Hi everyone, i am doing my first steps with FlowiseAI. I have problems to vectorize to JSON Files us. I use JSON File to upload a file. I have also found out, that i had to add a jq_schema as additional parameter. I used jq on the command line so now the documents looks like am i expecting this. The Array has been split into individual documents as in this example.

```
{
  "title": "Einführung ins Töpfern für Anfänger",
  "course_number": 101,
  "date": "03. Januar 2024",
  "description": "Eine grundlegende Einführung ins Töpfern, bei der die Teilnehmer die Basiswerkzeuge und -techniken kennenlernen.",
  "participants": [
    "Anna Schmidt",
    "Bernd Müller"
  ]
}
{
  "title": "Töpferrad 101",
  "course_number": 102,
  "date": "10. Januar 2024",
  "description": "Die Grundlagen der Verwendung des Töpferrades, inklusive Zentrierung und Öffnung des Tons.",
  "participants": [
    "Christian Weber",
    "Diana Keller",
    "Erik Lang"
  ]
}
```

But when it comes to the splitter i had no idea what to use. I tried most of them with no luck. In my vector database i can see each single line with no other context. So the information how is participating in the course is lost or at least hidden.

So want is the intended way to split um an array into multiple vectors and keep context?

**Comment by mwmdev** [comment]
I'm facing the same issue.

Langchain has a [Recursive JSON Splitter](https://python.langchain.com/v0.1/docs/modules/data_connection/document_transformers/recursive_json_splitter/), maybe this will be added to Flowise.
